SU2_CFD/src/solvers/CFEASolver.cpp

```diff
@@ -687,6 +692,10 @@ void CFEASolver::Compute_StiffMatrix(CGeometry *geometry, CNumerics **numerics,
             su2double val_Sol = nodes->GetSolution(indexNode[iNode],iDim) + val_Coord;
             element->SetRef_Coord(iNode, iDim, val_Coord);
             element->SetCurr_Coord(iNode, iDim, val_Sol);
+          }  
+          if (heat_solver) {
+                        auto nodal_temperatures = dynamic_cast<CSolver *>(heat_solver)->GetNodalTemperature();
+                        element->SetTemperature(iNode, nodal_temperatures[indexNode[iNode]]);
```

Comment:
Temperature is the solution of the heat solver, the "solution" (and other fields) are stored in CVariable classes which are the "nodes" member of the solvers.
```suggestion
            const su2double temperature = heat_solver->GetNodes()->GetSolution(indexNode[iNode], 0);
            element->SetTemperature(iNode, nodal_temperature);
```